app/invocation/invocation_share_button.tsx

```diff
@@ -12,50 +14,80 @@ import Dialog, {
 import Input from "../components/input/input";
 import Modal from "../components/modal/modal";
 import Select, { Option } from "../components/select/select";
+import rpcService from "../service/rpc_service";
 import InvocationModel from "./invocation_model";
 
 export interface InvocationShareButtonComponentProps {
   user?: User;
   model: InvocationModel;
+  invocationId: string;
 }
 
 interface State {
   isOpen: boolean;
-  // TODO: Replace with request proto once it's available.
-  visibility?: VisibilitySelection;
+  isLoading: boolean;
+  acl: acl.IACL;
+  error?: string;
 }
 
-type VisibilitySelection = "owner" | "group" | "public";
+type VisibilitySelection = "group" | "public";
 
 export default class InvocationShareButtonComponent extends React.Component<
   InvocationShareButtonComponentProps,
   State
 > {
-  state: State = { isOpen: false };
-
   private inputRef = React.createRef<HTMLInputElement>();
 
-  componentDidMount() {
-    // TODO: Fetch current invocation permissions
+  constructor(props: InvocationShareButtonComponentProps) {
+    super(props);
+    this.state = this.getInitialState();
+  }
+
+  componentDidUpdate(prevProps: InvocationShareButtonComponentProps) {
+    if (prevProps.invocationId !== this.props.invocationId) {
+      this.state = this.getInitialState();
+    }
   }
 
-  private onClick() {
+  private getInitialState(): State {
+    return { isOpen: false, acl: this.getInvocation().acl, isLoading: false };
+  }
+
+  private getInvocation() {
+    return this.props.model.invocations.find((invocation) => invocation.invocationId === this.props.invocationId);
+  }
+
+  private onShareButtonClick() {
     this.setState({ isOpen: true });
   }
 
   private onRequestClose() {
     this.setState({ isOpen: false });
   }
 
-  private onInputClick() {
+  private onLinkInputClick() {
     this.inputRef.current.select();
   }
 
-  private onVisibilityChange(e: React.ChangeEvent) {
-    const visibility = (e.target as HTMLSelectElement).value as VisibilitySelection;
-    this.setState({ visibility });
+  private async onVisibilitySelectionChange(e: React.ChangeEvent<HTMLSelectElement>) {
+    const visibility = e.target.value as VisibilitySelection;
+    const newAcl = new acl.ACL(this.getInvocation().acl);
+
+    newAcl.groupPermissions.read = visibility === "group" || visibility === "public";
```

Comment:
do we need to touch group permissions? can we just delete this line?

Reply:
Ah yep, this was left over from when I had private permissions as a possibility. Removed